src/main/java/logic/AdvancementLogic.java

```diff
@@ -0,0 +1,111 @@
+                .setRequired(false);
+        final SlashCommandOptionBuilder minimumFacets = new SlashCommandOptionBuilder()
+                .setType(SlashCommandOptionType.LONG)
+                .setName("minimum-facets")
+                .setDescription("Ensures as many specialty skills as possible are above the specified facet count, defaults to 12")
+                .setRequired(false);
+        final SlashCommandOptionBuilder nervewrightMana = new SlashCommandOptionBuilder()
+                .setType(SlashCommandOptionType.LONG)
+                .setName("nervewright-mana")
+                .setDescription("Specifies the amount of mana that can be invested for Nervewright, defaults to 0")
+                .setRequired(false);
+        final SlashCommandOptionBuilder adolescentInterests = new SlashCommandOptionBuilder()
+                .setType(SlashCommandOptionType.BOOLEAN)
+                .setName("adolescent-interests")
+                .setDescription("Set to true to factor in the effects of adolescent interests, defaults to false")
+                .setRequired(false);
+        final SlashCommandOptionBuilder nonEphemeral = new SlashCommandOptionBuilder()
+                .setType(SlashCommandOptionType.BOOLEAN)
+                .setName("non-ephemeral")
+                .setDescription("Set to true to send the message to the channel, defaults to false")
+                .setRequired(false);
+        options.add(SlashCommandOption.createWithOptions(SlashCommandOptionType.SUB_COMMAND, "tall", "Calculates the skill array to reach a given general skill by going tall.", targetFacets, startingArray, targetExpert, minimumFacets, nervewrightMana, adolescentInterests, nonEphemeral));
+        options.add(SlashCommandOption.createWithOptions(SlashCommandOptionType.SUB_COMMAND, "wide", "Calculates the skill array to reach a given general skill facet as cheaply as possible.", targetFacets, startingArray, targetExpert, nervewrightMana, adolescentInterests, nonEphemeral));
+        VelenCommand.ofSlash("advancement", "Calculates the AP cost to get a general skill", velen, new AdvancementLogic())
+                .addOptions(options.toArray(new SlashCommandOption[]{}))
+                .setServerOnly(true, 468046159781429250L)
+                .attach();
+    }
+
+    @Override
+    public void onEvent(SlashCommandInteraction event, User user, VelenArguments args, List<SlashCommandInteractionOption> options, InteractionImmediateResponseBuilder firstResponder) {
+        final SlashCommandInteractionOption subcommandOption = event.getOptions().get(0);
+        final Optional<Long> targetFacets = subcommandOption.getOptionLongValueByName("target-facets");
+        if (targetFacets.isPresent()) {
+            final Boolean expert = subcommandOption.getOptionBooleanValueByName("target-expert").orElse(false);
+            final List<Integer> startingArray = subcommandOption.getOptionStringValueByName("starting-array")
```

Comment:
Zero counts to make sure it's valid numbers, nor any recovery.

Reply:
Added validation for the params, may need more validation for the actual logic
![image](https://user-images.githubusercontent.com/11682762/145510392-7f700597-72b2-4f9b-bdea-fead91a7db5b.png)